code/game/gamemodes/setupgame.dm

```diff
@@ -120,7 +120,7 @@
 		if(G.block)
 			if(G.block in blocks_assigned)
 				warning("DNA2: Gene [G.name] trying to use already-assigned block [G.block] (used by [english_list(blocks_assigned[G.block])])")
-			dna_genes.Add(G)
+			dna_genes[G.type] = G

```

Comment:
What? Naw as far as I've seen you can use 
- Literally any object
- Strings
- Typepaths

Its highly compatible, I use typepaths pretty often when I can (see resetvars as an example)